sympy/core/equation.py

```diff
@@ -0,0 +1,359 @@
+    At present only the "=" relation operator is recognized.
+    
+    This class is intended to allow using the mathematical tools in SymPy to rearrange equations and perform algebra
+    in a stepwise fashion. In this way more people can successfully perform algebraic rearrangements without stumbling
+    over missed details such as a negative sign.
+    
+    __Note__ that this module imports Sympy into its namespace so there is no need to import Sympy separately.
+    
+    Create an equation with the call ``Equation(lhs,rhs,relation_operator)``, where ``lhs`` and ``rhs`` are any valid Sympy
+    expression. ``relation_operator`` defaults to the string "=" if not supplied. Currently,"=" is the only valid option.
+    ``Eqn(...)`` is a synonym for ``Equation(...)``.
+    
+    Examples
+    ========
+    >>> var('a b c')
+    >>> equ(a,b/c)
+    a=b/c
+    >>> t=equ(a,b/c)
+    >>> t
+    a=b/c
+    >>> t*c
+    a*c=b
+    >>> c*t
+    a*c=b
+    >>> exp(t)
+    exp(a)=exp(b/c)
+    >>> exp(log(t))
+    a=b/c
+
+    Integration can only be performed on one side at a time.
+    >>> q=Eqn(a*c,b/c)
+    >>> integrate(q,b,side='rhs')
+    b**2/(2*c)
+    >>> integrate(q,b,side='lhs')
+    a*b*c
+
+    >>> # Make a pretty statement of integration from an equation 
+    >>> Eqn(Integral(q.lhs,b),integrate(q,b,side='rhs'))                        
+    Integral(a*c, b)=b**2/(2*c)
+    >>> # This is duplicated by the convenience function self.integ 
+    >>> q.integ(b)                                                              
+    Integral(a*c, b)=b**2/(2*c)
+
+    SymPy's solvers do not understand these equations. They expect an expression that the solver assumes = 0.
+    Thus to use the solver the equation must be rearranged so that all non-zero symbols are on one side. Then
+    just the non-zero symbolic side is passed to ``solve()``.
+    >>> t2 = t-t.rhs
+    >>> t2
+    a-b/c=0
+    >>> solve(t2.lhs,c)
+    [b/a]
+    """
+
+    def __new__(cls, lhs, rhs, relop='='):
+        if not(relop == '='):
+           raise NotImplementedError('"=" is the only relational operator presently supported in Equations.')
+        lhs = _sympify(lhs)
+        rhs = _sympify(rhs)
+        if (lhs.is_number) and (rhs.is_number) and not (lhs == rhs):
+           print('WARNING: did your really mean to define unequal numbers as equal? ' + str(lhs) +'='+ str(rhs))
+        return super().__new__(cls, lhs, rhs, relop)
+
+    @property
+    def lhs(self):
+        """
+        Returns the lhs of the equation.
+        """
+        return self.args[0]
+
+    @property
+    def rhs(self):
+        """
+        Returns the rhs of the equation.
+        """
+        return self.args[1]
+    
+    @property
+    def relop(self):
+        """
+        Returns the string representing the relationship operator.
+        """
+        return self.args[2]
```

Comment:
I don't think we should store the operation type in the args. If we want to support other types of relationals they should have separate classes, just as is done with the boolean relationals. I would expect `lhs, rhs = eq.args` to work.

Reply:
Taking out `relop`. Will resolve when committed.